src/main/scala/org/broadinstitute/dsde/workbench/leonardo/Boot.scala

```diff
@@ -8,13 +8,17 @@ import com.typesafe.scalalogging.LazyLogging
 import net.ceedubs.ficus.Ficus._
 import org.broadinstitute.dsde.workbench.leonardo.api.LeoRoutes
 import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.dns.ClusterDnsCache
 import org.broadinstitute.dsde.workbench.leonardo.config.{DataprocConfig, ProxyConfig, SwaggerConfig}
 import org.broadinstitute.dsde.workbench.leonardo.dao.GoogleDataprocDAO
 import org.broadinstitute.dsde.workbench.leonardo.service.{LeonardoService, ProxyService}
 
 object Boot extends App with LazyLogging {
   private def startup(): Unit = {
 
+    System.setProperty("sun.net.spi.nameservice.provider.1", "dns,Jupyter")
+    System.setProperty("sun.net.spi.nameservice.provider.2", "dns,sun")
```

Comment:
can you comment some about who uses these?

Reply:
It sets the DNS providers in priority order. Top priority is our "fake" DNS server which resolves jupyter.firecloud.org hostnames to real IPs. (The "dns,Jupyter" matches the `JupyterNameServiceDescriptor.getType/getProviderName`.) If that doesn't resolve a given hostname, it falls back to the default Java DNS.

Reply:
Yeah, I'll add some comments